core/bin/block_reverter/src/main.rs

```diff
@@ -103,35 +122,111 @@ async fn main() -> anyhow::Result<()> {
+            decode_yaml_repr::<zksync_protobuf_config::proto::contracts::Contracts>(&yaml)
+                .context("failed decoding contracts YAML config")?
+        }
+        None => ContractsConfig::from_env().context("ContractsConfig::from_env()")?,
+    };
+    let secrets_config = if let Some(path) = opts.secrets_path {
+        let yaml = std::fs::read_to_string(&path).with_context(|| path.display().to_string())?;
+        let config = decode_yaml_repr::<zksync_protobuf_config::proto::secrets::Secrets>(&yaml)
+            .context("failed decoding secrets YAML config")?;
+        Some(config)
+    } else {
+        None
+    };
+
     let default_priority_fee_per_gas = eth_sender
         .gas_adjuster
         .context("gas_adjuster")?
         .default_priority_fee_per_gas;
-    let contracts = ContractsConfig::from_env().context("ContractsConfig::from_env()")?;
-    let network = NetworkConfig::from_env().context("NetworkConfig::from_env()")?;
-    let database_secrets = DatabaseSecrets::from_env().context("DatabaseSecrets::from_env()")?;
-    let l1_secrets = L1Secrets::from_env().context("L1Secrets::from_env()")?;
-    let postgress_config = PostgresConfig::from_env().context("PostgresConfig::from_env()")?;
-    let era_chain_id = env::var("CONTRACTS_ERA_CHAIN_ID")
-        .context("`CONTRACTS_ERA_CHAIN_ID` env variable is not set")?
-        .parse()
-        .map_err(|err| {
-            anyhow::anyhow!("failed parsing `CONTRACTS_ERA_CHAIN_ID` env variable: {err}")
-        })?;
-    let config = BlockReverterEthConfig::new(&eth_sender, &contracts, &network, era_chain_id)?;
+
+    let database_secrets = match &secrets_config {
+        Some(secrets_config) => secrets_config
+            .database
+            .clone()
+            .context("Failed to find database config")?,
+        None => DatabaseSecrets::from_env().context("DatabaseSecrets::from_env()")?,
+    };
+    let l1_secrets = match &secrets_config {
+        Some(secrets_config) => secrets_config
+            .l1
+            .clone()
+            .context("Failed to find l1 config")?,
+        None => L1Secrets::from_env().context("L1Secrets::from_env()")?,
+    };
+    let postgres_config = match &general_config {
+        Some(general_config) => general_config
+            .postgres_config
+            .clone()
+            .context("Failed to find postgres config")?,
+        None => PostgresConfig::from_env().context("PostgresConfig::from_env()")?,
+    };
+    let network_config = match &genesis_config {
+        Some(genesis_config) => NetworkConfig {
+            network: Network::from_chain_id(genesis_config.l1_chain_id),
+            zksync_network: genesis_config.l2_chain_id.as_u64().to_string(),
+            zksync_network_id: genesis_config.l2_chain_id,
+        },
+        None => NetworkConfig::from_env().context("NetworkConfig::from_env()")?,
+    };
+
+    let config = BlockReverterEthConfig::new(&eth_sender, &contracts, &network_config)?;
```

Comment:
could we reduce this dependency? we need only `zksync_network_id` in `BlockReverterEthConfig`

Reply:
In order to reduce the dependency, I have to manually try to load the `zksync_network_id` from env, which doesn't seem like the correct approach. With this criteria `NetworkConfig` was removed from `BlockReverterEthConfig` but not as a dependency for `block_reverter` bin:

https://github.com/matter-labs/zksync-era/blob/f3447ff358e1c0afe2eacd6fea1baff3cd7b55aa/core/bin/block_reverter/src/main.rs#L209-L218